README.md

```diff
@@ -289,7 +289,7 @@ const nc = await connect({ url: "demo.nats.io:4222" });
 // a client makes a request and receives a promise for a message
 // by default the request times out after 1s (1000 millis) and has
 // no payload.
-await nc.request("time", 1000, "hello!")
+await nc.request("time", "hello!", { timeout: 1000 })
```

Comment:
Why though? Meaning you did not address that in the summary of the PR: why is timeout an option now? Requesting without a timeout sounds to me like a publish.

Reply:
the timeout is there as an option, this way we can have other options like headers - the timeout is set by default to 1000, whether it is provided or not.

Reply:
Also thinking of moving the message payload as part of the option for publish/request. The reason being is that the default arguments work well until you need to specify one of them. For request, the only required argument is the subject, the rest are defaulted, same for publish. This simplifies the syntax, as now you can specify an optional value, without having to put a dummy value on the previous optional argument:

`publish/request('foo', {headers: h})` vs. `publish/request('foo', '', {headers: h})` in the case where the payload is not a string, it becomes more problematic. In the case of the request, the timeout will always be set to a default of 1000, unless client provides a different value.

Reply:
Yes, but if payload is optional, what will be the syntax when user wants to provide it? Does that becomes `publish('foo', {payload: 'bar', headers: h})`? So in more common situations (presence of a payload), there is more to "type" if you will.

Reply:
yeap.